virttest/utils_disk.py

```diff
@@ -839,8 +840,15 @@ def set_drive_letter(session, did, partition_no=1):
         if re.search("Reserved", line, re.I | re.M):
             partition_no += int(line.split()[1])
     assign_letter_cmd = ' echo select partition %s ; echo assign '
+    if target_letter:
+        assign_letter_cmd += 'letter=%s' % target_letter
     assign_letter_cmd = _wrap_windows_cmd(assign_letter_cmd)
     session.cmd(assign_letter_cmd % (did, partition_no))
+    if target_letter:
+        if target_letter != get_drive_letters(session, did):
+            raise exceptions.TestError("Not assign the target letter '%s' "
```

Comment:
Btw, according to the original design it should return `None` in this situation.